src/figuresClasses.ts

```diff
@@ -1,19 +1,81 @@
-export interface Figure {
-
+/* eslint-disable no-useless-constructor */
+enum Color {
+  red = 'red',
+  green = 'green',
+  blue = 'blue',
 }
 
-export class Triangle {
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
+}
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if ((a <= 0 || b <= 0 || c <= 0)
+      || ((a + b) <= c || (a + c) <= b || (b + c) <= a)
+    ) {
+      throw new Error('invalid data');
+    }
```

Comment:
we can put some of this into a separate variable or break it into two if`s for better readability